src/core/lib/security/credentials/google_default/google_default_credentials.cc

```diff
@@ -80,21 +81,22 @@ grpc_google_default_channel_credentials::create_security_connector(
     grpc_core::RefCountedPtr<grpc_call_credentials> call_creds,
     const char* target, const grpc_channel_args* args,
     grpc_channel_args** new_args) {
-  bool is_grpclb_load_balancer = grpc_channel_arg_get_bool(
-      grpc_channel_args_find(args, GRPC_ARG_ADDRESS_IS_GRPCLB_LOAD_BALANCER),
-      false);
-  bool is_backend_from_grpclb_load_balancer = grpc_channel_arg_get_bool(
-      grpc_channel_args_find(
-          args, GRPC_ARG_ADDRESS_IS_BACKEND_FROM_GRPCLB_LOAD_BALANCER),
-      false);
-  bool use_alts =
-      is_grpclb_load_balancer || is_backend_from_grpclb_load_balancer;
+  const bool is_grpclb_load_balancer = grpc_channel_args_find_bool(
+      args, GRPC_ARG_ADDRESS_IS_GRPCLB_LOAD_BALANCER, false);
+  const bool is_backend_from_grpclb_load_balancer = grpc_channel_args_find_bool(
+      args, GRPC_ARG_ADDRESS_IS_BACKEND_FROM_GRPCLB_LOAD_BALANCER, false);
+  const char* xds_cluster =
+      grpc_channel_args_find_string(args, GRPC_ARG_XDS_CLUSTER_NAME);
+  const bool is_xds_cfe_cluster =
+      xds_cluster != nullptr && strcmp(xds_cluster, "google_cfe") == 0;
+  const bool use_alts = is_grpclb_load_balancer ||
+                        is_backend_from_grpclb_load_balancer ||
+                        is_xds_cfe_cluster;
```

Comment:
The "google_cfe" cluster is for CFE endpoints only, right?

If so, it seems incorrect to set `use_alts = true` in this case, since we can't use ALTS to talk to the CFEs

Reply:
Doh!  You're right, I had this condition backwards.  Fixed now.

Thanks for catching this!  When we set up the C2P interop tests, we'll need to make sure we cover this case.

Reply:
Btw, this is also flipped in our tracking spread sheet and the _Work Needed_ section of the design doc. 😄 (Yeah, it is a bit confusing)

Reply:
Thanks for pointing this out.  I've fixed it in both places.